### PR TITLE
Reset value of input field of SearchWidget.

### LINKED ISSUE #4028: Input value of Serach field remains even after hitting submit button
## Issue
When the input is given inside the search field the value remains there even if we change our routes or do editing etc. which I think does not look so good.
This video illustrates the above issue:

https://user-images.githubusercontent.com/99530996/205440822-41bf8e6b-dea0-4112-99e3-5b874f29f9cf.mp4

## Approach

1. In `plone/volto/src/components/theme/SearchWidget/SearchWidget.jsx` we can modify the handler `onSubmit()` to reset the value of the input field of the search widget.
OR
2. As proposed by tiberiuichim#7386 on discord we can replace the input field with a button that takes you to the search page just like in bbc.com.

https://user-images.githubusercontent.com/99530996/205442462-cb27c76c-0d67-48fe-aecc-e159f4c1b076.mp4

Feedback Welcome.

### PR DESCRIPTION
**Pull request by MAX-786**
fixes #4028 

## Approach
reset the value of text using setState() :
```js
  onSubmit(event) {
    const path =
      this.props.pathname?.length > 0
        ? `&path=${encodeURIComponent(this.props.pathname)}`
        : '';
    this.props.history.push(
      `/search?SearchableText=${encodeURIComponent(this.state.text)}${path}`,
    );
 // *** reset value of input field
    this.setState({
      text: '',
    });
 // ***
    event.preventDefault();
  }

```
## Demo:

https://user-images.githubusercontent.com/99530996/206274415-094fda89-88e9-4254-b48a-b66d48cad208.mp4

### OS: Any

Feedback Welcome.

### CODE REVIEW

File: src/components/theme/SearchWidget/SearchWidget.jsx [hunk 1]
```diff
@@ -81,6 +81,10 @@ class SearchWidget extends Component {
     this.props.history.push(
       `/search?SearchableText=${encodeURIComponent(this.state.text)}${path}`,
     );
+    // reset input value
```
Comment (davisagli) [inline src/components/theme/SearchWidget/SearchWidget.jsx:84]:
```suggestion
    // reset input value
```

The lint check doesn't like the extra space at the end of the line.

File: news/4028.bugfix [hunk 2]
```diff
@@ -0,0 +1 @@
+reset value of search field after submit. [@MAX-786]
```
Comment (davisagli) [inline news/4028.bugfix:1]:
```suggestion
Reset value of search field after submit. [@MAX-786]
```

**Review by davisagli — APPROVED** [review]

**Comment by davisagli** [comment]
@MAX-786 I already approved it but I am leaving it for @sneridagh to merge once he is back from holiday.

**Comment by MAX-786** [comment]
Thanks, @davisagli for reviewing

**Review by sneridagh — APPROVED** [review]

**Pull request merged**